esx_service/vmdk_ops.py

```diff
@@ -648,10 +649,10 @@ def disk_detach(vmdk_path, vm):
 
     try:
         wait_for_tasks(si, [vm.ReconfigVM_Task(spec=spec)])
-    except vim.fault.GenericVmConfigFault as ex:
-        for f in ex.faultMessage:
-            logging.warning(f.message)
-        return err("Failed to detach " + vmdk_path)
+    except vim.Fault.VimFault as ex:
+        msg = "Failed to detach %s (%s)" % (vmdk_path, ex.msg)
+        logging.warning(msg + "\n" + str(ex))
+        return err(msg)
 
```

Comment:
This will leave the disk attached to the VM. Like I'd mentioned in PR #433 (original set of changes) there is nothing the plugin can do to help this situation except log the error. Does the error notify docker that the unmount has failed and docker must keep track that the volume is still in use? The volume remains in attached state if docker doesn't care to unmount again.